Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -157,13 +165,78 @@ public void Process(ITelemetry telemetry)
+            {
+                return new RequestTelemetryDocument()
+                           {
+                               Version = TelemetryDocumentContractVersion,
+                               Timestamp = telemetryAsRequest.Timestamp,
+                               Id = telemetryAsRequest.Id,
+                               Name = telemetryAsRequest.Name,
+                               StartTime = telemetryAsRequest.StartTime,
+                               Success = IsRequestSuccessful(telemetryAsRequest),
+                               Duration = telemetryAsRequest.Duration,
+                               Sequence = telemetryAsRequest.Sequence,
+                               ResponseCode = telemetryAsRequest.ResponseCode,
+                               Url = telemetryAsRequest.Url,
+                               HttpMethod = telemetryAsRequest.HttpMethod
+                           };
+            }
+            else if (telemetryAsDependency != null)
+            {
+                return new DependencyTelemetryDocument()
+                           {
+                               Version = TelemetryDocumentContractVersion,
+                               Timestamp = telemetryAsDependency.Timestamp,
+                               Id = telemetryAsDependency.Id,
+                               Name = telemetryAsDependency.Name,
+                               StartTime = telemetryAsDependency.StartTime,
+                               Success = telemetryAsDependency.Success,
+                               Duration = telemetryAsDependency.Duration,
+                               Sequence = telemetryAsDependency.Sequence,
+                               ResultCode = telemetryAsDependency.ResultCode,
+                               CommandName = telemetryAsDependency.CommandName,
+                               DependencyTypeName = telemetryAsDependency.DependencyTypeName,
+                               DependencyKind = telemetryAsDependency.DependencyKind
+                           };
+            }
+            else if (telemetryAsException != null)
+            {
+                // //!!! cut length for Exception.ToString()
+                return new ExceptionTelemetryDocument()
+                           {
+                               Version = TelemetryDocumentContractVersion,
+                               Message = telemetryAsException.Message,
+                               SeverityLevel =
+                                   telemetryAsException.SeverityLevel != null
+                                       ? telemetryAsException.SeverityLevel.Value.ToString()
+                                       : null,
+                               HandledAt = telemetryAsException.HandledAt.ToString(),
+                               Exception =
+                                   telemetryAsException.Exception != null
+                                       ? telemetryAsException.Exception.ToString()
```

Comment:
We'll work on this in the future once e2e is done. I've looked at ToInvariantString and wasn't sure what it was doing exactly, I'll get back to you on that.